src/tokens.ts

```diff
@@ -550,6 +550,16 @@ export interface IGitExtension extends IDisposable {
    */
   showTopLevel(path: string): Promise<string | null>;
 
+  /**
+   * Stash the current changes in a dirty repository.
+   * @param path - path at which the stashes will be saved in
+   * @returns promise which resolves upon stashing changes
+   * @throws {Git.NotInRepository} If the current path is not a Git repository   * @throws {Git.GitResponseError} If the server response is not ok
+   * @throws {ServerConnection.NetworkError} If the request cannot be made
+   *
+   */
+  stashChanges(path: string): Promise<void>;
+
```

Comment:
You should add in this interface 

```ts
readonly stashChanged: ISignal<IGitExtension, IChangedArgs<Git.IStash>>
```

this may be related to some errors you are seeing with undefined `stashChanged`